opentelemetry-sdk/src/metrics/attribute_set.rs

```diff
@@ -0,0 +1,55 @@
+use std::hash::{Hash, Hasher};
+
+use opentelemetry::{Key, KeyValue, Value};
+use rustc_hash::FxHasher;
+
+/// A unique set of attributes that can be used as instrument identifiers.
+///
+/// This must implement [Hash], [PartialEq], and [Eq] so it may be used as
+/// HashMap keys and other de-duplication methods.
+#[derive(Clone, Default, Debug, PartialEq, Eq)]
+pub(crate) struct AttributeSet(pub(crate) Vec<KeyValue>, u64);
```

Comment:
its not clear what is the reason why this is being changed? can you update the PR desc. with the reasoning. 
if the goal is to get predictable attribute set in readers, then https://github.com/open-telemetry/opentelemetry-rust/pull/2093 looks sufficient to achieve that, without having to touch the hot path code.